src/qibolab/instruments/qm/program/acquisition.py

```diff
@@ -111,8 +114,8 @@ def fetch(self, handles):
         qres = handles.get(f"{self.name}_Q").fetch_all()
         # convert raw ADC signal to volts
         u = unit()
-        signal = collect(u.raw2volts(ires), u.raw2volts(qres))
-        return _split(signal, self.npulses, iq=True)
+        signal = _collect(u.raw2volts(ires), u.raw2volts(qres), self.npulses)
+        return _split(signal, self.npulses)
```

Comment:
Do you really need to *collect **and** split*?

It's a real question, I'm not sure: they seem to be the inverse operations, but it's true that they are saving you inlining explicit conditionals and less intuitive calls to `np.moveaxis` (which are encapsulated in more meaningful operations)

Reply:
I tried a bit to work around them but I couldn't find any way that I like more.

They are not really inverse, even if the name suggests so. `_collect` collects the I and Q components which are saved in separate streams and downloaded as separate arrays from the instrument. `_split` splits acquisitions beloning to different measurement pulses which are mapped to the same acquisition stream so they are downloaded in a single array. They are needed because qibolab expects an array per measurement pulse, which contains both I and Q.

Maybe we can eliminate them by refactoring the streams, but since it's fully internal I didn't spend much time on it.

Reply:
Thanks for the explanation, I agree with the conclusion.